docs/agentless.md

```diff
@@ -0,0 +1,64 @@
+# PingFederate Authentication API Agentless Integration
+
+### Overview
+You can modify the JavaScript Widget to allow the Agentless Integration Kit to work with the PingFederate Authentication API.
+
+### System requirements and dependencies
+* Agentless Integration Kit 2.0 or later
+* An authentication service that is configured to drop off authenticated user attributes to PingFederate
+
+### Setup
+
+1. Modify the `reference_id_required.hbs` template to include the credential fields that your authentication service requires. The widget renders this handlebar template when it encounters the `REFERENCE_ID_REQUIRED` state.
+
+1. Add an event handler that sends the user credentials to the authentication service. The event handler is triggered when the user submits the form on the template.
+
+	Example constructor additions:
+	```js
```

Comment:
What do you guys think about just having this code in the widget and the instructions just reference filling out the empty `handleAgentlessSignOn` method? 

I _think_ it would be harmless to have this code in there already.

Reply:
I think it's a good idea. we should provide proper console log messages to guide developers toward proper implementation.

Reply:
I think it would be harmless as well, I was just under the impression that we wanted to limit the _useless_ code in the widget unless someone wanted to integrate with agentless. I can add the code in the from the code blocks to the `index.js` and add in some console logging for `registerAgentlessHandler()` and `handleAgentlessSignOn(evt)`.